lib/request-wrapper.ts

```diff
@@ -306,20 +307,17 @@ export class RequestWrapper {
 
       error.message = parseServiceErrorMessage(axiosError.data) || axiosError.statusText;
 
-      // some services bury the useful error message within 'data'
-      // adding it to the error under the key 'body' as a string or object
-      let errorBody;
+      // attach the error response body to the error
       try {
-        // try/catch to handle objects with circular references
-        errorBody = JSON.stringify(axiosError.data);
+        // try/catch to detect objects with circular references
+        JSON.stringify(axiosError.data);
       } catch (e) {
-        // ignore the error, use the object, and tack on a warning
-        errorBody = axiosError.data;
-        errorBody.warning = 'Body contains circular reference';
-        logger.error(`Failed to stringify axiosError: ${e}`);
+        logger.warn('Error field `result` contains circular reference(s)');
+        logger.debug(`Failed to stringify error response body: ${e}`);
       }
 
-      error.body = errorBody;
+      error.result = ensureJSONResponseBodyIsObject(axiosError);
+      error.body = error.result; // ** deprecated **
```

Comment:
Putting the `object` form of the response into the `result` field to match success cases opens an opportunity to reduce the impact of this otherwise breaking change when the type becomes an `object` instead of a `string`. I know, for instance, that in our SDK we have the `error.body` documented as "the error response body as text".

As such I wonder if it is worth leaving the deprecated `body` property as a `string` type i.e. `error.body = axiosError.data;` to preserve the behaviour for existing code rather than having 2 copies of the `object` form?

Reply:
Ah that is a great idea, thanks for the suggestion! I will do that 👍

Reply:
Reverted `error.body` to previous behavior and updated the tests